src/packet_macros.rs

```diff
@@ -10,23 +10,13 @@ macro_rules! packet_boilerplate {
             $($name($name)),*
         }
 
-        pub fn read<S: MinecraftProtocolReader + Read>(stream: &mut S, state: i32, length: i32) -> Packet {
-            //Read the entire packet into a vector first
-            let vec: Vec<u8> = stream
-                .bytes()
-                .take(length as usize)
-                .map(|r: Result<u8, _>| r.expect("packet was smaller than length field indicated!"))
-                .collect();
-            let mut cursor = Cursor::new(vec);
```

Comment:
I've moved this part into the server. I think that extracting bytes from the stream probably does belong to the server

It's also convenient from a technical perspective to not be passing around tcpstreams but small chunks of data for single packets